File: java/src/main/java/com/company/automation/models/Category.java
package com.company.automation.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/** A product category. */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Category(
    @JsonProperty("id") String id,
    @JsonProperty("name") String name,
    @JsonProperty("slug") String slug) {}

File: csharp/src/AutomationFramework.Core/Models/Category.cs
using System.Text.Json.Serialization;

namespace AutomationFramework.Core.Models;

/// <summary>
/// A product category.
/// </summary>
/// <remarks>
/// <c>ParentId</c> appears only on <c>GET /products/{id}</c>, where categories are expanded with their
/// parent, and is null for a top-level category. The list endpoint omits it entirely.
/// </remarks>
public sealed record Category(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("slug")] string Slug,
    [property: JsonPropertyName("parent_id")] string? ParentId);
